src/data_snack/entities/validation.py

```diff
@@ -87,3 +89,63 @@ def validate_meta_sources_fields(entity_class: ABCMeta) -> None:
         for source in entity_class.Meta.sources
     ])):
         raise SourceEntityFieldException(f"Missing source entity fields: {missing_mappings}.")
+
+
+def validate_unique_fields_between_source_entities(entity_class: ABCMeta) -> None:
+    """
+    Validates if there are no duplicated fields between source entities except their keys.
+    :param entity_class: entity to validate
+    """
+    if duplicated_fields := [
```

Comment:
Consider using normal for loops for those functions. It may be a bit more readable than functional programming (and, according to profiler, is almost 2x faster):

```
def validate_unique_fields_between_source_entities_2(entity_class) -> None:
    """
    Validates if there are no duplicated fields between source entities except their keys.
    :param entity_class: entity to validate
    """
        
    fields = {}
    for source in entity_class.Meta.sources:
        _source_keys = source.entity.get_keys()
        for mapping in source.entity_fields_mapping:
            if mapping.source_field not in _source_keys:
                fields[mapping.field] = fields.get(mapping.field, 0) + 1

    if duplicated_fields := [f for (f, c) in fields.items() if c > 1]:
        raise DuplicatedFieldsException(f"Duplicated fields between source entities: {duplicated_fields}")
    ```